compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -951,6 +951,18 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
 
                 if imm_result || mut_result {
                     if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(span) {
+                        // This if is to prevent a special edge-case
```

Comment:
Can we expand what the edge-case is? It might be fine (anyone curious can remove the code and re-tidy to check).

Reply:
The edge case is suggesting stuff like adding `&` to a non-expr (e.g. a field) due to a derive macro

```rust
struct Foo {
  &f: Bar,
  +
}
```

For the record I didn't write this comment, I just moved this `if` statement forward more since it makes more sense. It was introduced in dfbc143e65dd4dc8499f7296ddc7889854a8cc7d, which should've gated this whole block and not just the suggestion, imo.